### PR TITLE
Remove System.Collections.Immutable dependency from net6.0 target

### LINKED ISSUE #19: Provide additional TFM to enable 0 external dependencies
### Is your feature request related to a problem? Please describe.
I want to minimise dependencies in my project by utilising framework dependencies wherever possible

### Describe the solution you'd like
I want the package to not have a dependencies on packages which can be provided by the framework ie System.Collections.Immutable

### Describe alternatives you've considered
Accept the additional dependency

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by thompson-tomo**
Drop System.Collection.Immutable to v6 and make it conditional

Closes #19

### CODE REVIEW

File: src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj [hunk 1]
```diff
@@ -1,12 +1,14 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net6.0;net7.0;net8.0</TargetFrameworks>
```
Comment (AArnott) [inline src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj:3]: .NET 7 falls out of support next month. It hardly seems worthwhile to add support for it explicitly now. 
And anyway, it and net8.0 adds nothing that the pre-existing net6.0 target already offers. So let's revert the whole change.

File: src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Description>Algorithms such as branch and bound.</Description>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.BannedApiAnalyzers" PrivateAssets="all" />
     <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" PrivateAssets="all" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)'== 'netStandard2.0' or '$(TargetFramework)' == 'netStandard2.1'">
```
Comment (AArnott) [inline src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj:11]: Nothing here targets netstandard2.1. Let's drop the condition.
And let's stick with canonical capitalization for `netstandard2.0` please.

File: global.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "7.0.101",
+    "version": "8.0.204",
```
Comment (AArnott) [inline global.json:3]: No need to upgrade this. 
I upgrade this by merging in the Library.Template repo periodically, so doing this directly here would just introduce merge conflicts.

**Review by AArnott — CHANGES_REQUESTED** [review]
Thanks.

File: src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj (hunk 1 above)
Comment (thompson-tomo) [reply src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj:3]: The issue which could arise is if, the condition below has the net6.0 removed then we could potentially be forcing a downgrade of System.Collections.Immutable to v6 and then some people would view that as a breaking change. If you like me don't see that as a breaking change, then i can drop net 7 & 8, the sdk upgrade and adjust the condition

Comment (AArnott) [reply src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj:3]: Oh, you're saying that the netstandard2.0 build will want System.Collections.Immutable 7.0, but the net6.0 build will want the 6.0 version of that assembly? That's very subtle, but interesting now that you've pointed it out. I suppose the right fix for that would be to change to never ask for more than 6.0 of the package, even for the netstandard2.0. So can we proceed with my request, but also change Directory.Packages.props to reduce the version referenced to 6.0.0?

Comment (thompson-tomo) [reply src/Nerdbank.Algorithms/Nerdbank.Algorithms.csproj:3]: Sure will drop dependency to v6 & run with net 6 as the TFM not net 7

**Review by AArnott — APPROVED** [review]
Thanks

**Pull request merged**